### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
- XMLDoc fix (revert fbecb5ee) for Persalys
- in VertexValuePointToFieldFunction, reuse input description of inner function, for otfmi
- Fix multistart evaluation number
- Add optimization status accessor:  getStatus()->int (SUCCEED/FAILURE/...?), getStatusMessage()->str for details
- Deprecate Sample|CorrelationAnalysis.computePearsonCorrelation #2477 
- Function.getCallsNumber now gives only direct calls #1252 
- Deprecated OptimizationAlgo.setMaximumEvaluation #2479 
- Add OptimizationAlgo.setCheckStatus for all algos to disable throwing (replaces TNC/Cobyla.setIgnoreFailure)#2460 
- Add max time for optim & simulation algos
- #2486
- promote uniformovermesh,integ/LSexpansion classes out of experimental
- #2503
- legend outside graph by default when plotting 2-d functions (PDF, CDF, ...)
- Drop DiscreteDistribution "computePDF(Indices)" signature in causing problems with swig 4.2 (and only actually used for Multinomial)
- fix indicescollection conversion from numpy int array
- allow to use Python limited api: https://docs.python.org/3/c-api/stable.html#limited-c-api
- fix incompletegamma for large x
- drop SpecFunc.isnan/isinf: #2511 
- retrieve LARS error history:
```
view = View(chaosResult.drawErrorHistory())
```
chaos_error

### CODE REVIEW

**Comment by sofianehaddad** [comment]
+1

File: python/src/KarhunenLoeveP1Algorithm_doc.i.in [hunk 1]
```diff
@@ -12,9 +12,12 @@ s : float, :math:`\geq0`
 
 Notes
 -----
-The Karhunen-Loeve :math:`P_1` algorithm solves the Fredholm problem  associated to the covariance function :math:`C`: see :class:`~openturns.KarhunenLoeveAlgorithm` to get the notations.
+The Karhunen-Loeve :math:`P_1` algorithm solves the Fredholm problem associated to the
+covariance function :math:`C`: see :class:`~openturns.KarhunenLoeveAlgorithm` to get the notations.
```
Comment (josephmure) [inline python/src/KarhunenLoeveP1Algorithm_doc.i.in:16]: for the notations

File: python/src/KarhunenLoeveP1Algorithm_doc.i.in [hunk 2]
```diff
@@ -12,9 +12,12 @@ s : float, :math:`\geq0`
...
 
-The Karhunen-Loeve :math:`P_1` approximation uses the :math:`P_1` functional basis :math:`(\theta_p)_{1 \leq p \leq N}` where  :math:`\theta_p: \cD_N \mapsto \Rset` are the basis functions of the :math:`P_1` finite element space associated to :math:`\cD_N`, which vertices are :math:`(\vect{s}_i)_{1 \leq i \leq N}`.
+The Karhunen-Loeve :math:`P_1` approximation uses the :math:`P_1` functional basis :math:`(\theta_p)_{1 \leq p \leq N}`
+where :math:`\theta_p: \cD_N \mapsto \Rset` are the basis functions of the :math:`P_1`
+finite element space associated to :math:`\cD_N`, which vertices are :math:`(\vect{s}_i)_{1 \leq i \leq N}`.
```
Comment (josephmure) [inline python/src/KarhunenLoeveP1Algorithm_doc.i.in:20]: whose

File: python/src/KarhunenLoeveP1Algorithm_doc.i.in [hunk 3]
```diff
@@ -31,7 +34,10 @@ The Galerkin approach and the collocation one are equivalent in the :math:`P_1`
 
 where :math:`\mat{G} = (G_{ij})_{1\leq i,j \leq N}` with :math:`G_{i\ell}= \int_{\cD} \theta_i(\vect{s})\theta_\ell(\vect{s})\,  d\vect{s}`, :math:`\mat{\Lambda}=diag(\vect{\lambda})`.
 
-Though the eigenvalues computation is performed by LAPACK by default, several other solvers can be used. The solver to use must be specified by setting ``KarhunenLoeveP1Algorithm-EigenvaluesSolver`` key in :class:`~openturns.ResourceMap`. Some solvers set limits to the number of eigenvalues they can compute with regards to the size of the problem. The following table summarizes the supported solvers and their maximum number of eigenvalues for a problem of size :math:`n`: 
+Though the eigenvalues computation is performed by LAPACK by default, several other solvers can be used.
+The solver to use must be specified by setting ``KarhunenLoeveP1Algorithm-EigenvaluesSolver`` key in :class:`~openturns.ResourceMap`.
```
Comment (josephmure) [inline python/src/KarhunenLoeveP1Algorithm_doc.i.in:38]: the

File: python/src/KarhunenLoeveP1Algorithm_doc.i.in [hunk 4]
```diff
@@ -41,7 +47,10 @@ Though the eigenvalues computation is performed by LAPACK by default, several ot
 | Spectra     | SPECTRA             | :math:`n-2`          |
 +-------------+---------------------+----------------------+
 
-For large dimension problems, it can be useful to generate the covariance matrix of the problem as a :class:`~openturns.HMatrix` instead of a standard dense matrix. In this case, one can set ``KarhunenLoeveP1Algorithm-CovarianceMatrixStorage`` key in :class:`~openturns.ResourceMap` to ``HMAT`` (instead of ``DENSE``).
+For large dimension problems, it can be useful to generate the covariance matrix of the
+problem as a :class:`~openturns.HMatrix` instead of a standard dense matrix.
+In this case, one can set ``KarhunenLoeveP1Algorithm-CovarianceMatrixStorage``
```
Comment (josephmure) [inline python/src/KarhunenLoeveP1Algorithm_doc.i.in:52]: the

**Comment by josephmure** [comment]
I was currently reviewing the PR.

File: lib/src/Base/Optim/OptimizationResult.cxx [hunk 5]
```diff
@@ -303,14 +336,31 @@ void OptimizationResult::store(const Point & x,
...
+      if (!SpecFunc::IsNormal(y[j]))
+        isNormal = false;
+
     const Bool objectiveImproved = (!getOptimalValue().getDimension())
                                    || ((getProblem().isMinimization() && y[0] < getOptimalValue()[0]) || (!getProblem().isMinimization() && y[0] > getOptimalValue()[0]));
 
     const Bool insideBounds = (!getProblem().hasBounds()) || (getProblem().hasBounds() && getProblem().getBounds().contains(x));
 
-    if ((objectiveImproved && insideBounds && (constraintError <= maximumConstraintError))
+    if ((isNormal && objectiveImproved && insideBounds && (constraintError <= maximumConstraintError))
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Base/Optim/OptimizationResult.cxx:363]: we dont want to allow nans, but inf is allowed only for the ouputs

**Pull request merged**